neptune/internal/threads/neptune_thread.py

```diff
@@ -15,15 +15,27 @@
 #
 import threading
 
+from neptune import utils
+
 
 class NeptuneThread(threading.Thread):
     def __init__(self, is_daemon):
         super(NeptuneThread, self).__init__(target=self.run)
         self.setDaemon(is_daemon)
         self._interrupted = threading.Event()
 
-    def is_interrupted(self):
-        return self._interrupted.is_set()
+    def should_continue_running(self):
+        # TODO: remove this pylint exception once we stop supporting Python 2
+        # pylint: disable=no-member
+        if utils.is_python_2():
+            all_threads = threading.enumerate()
+
+            # pylint:disable=protected-access
+            main_thread_is_alive = any(t.__class__ is threading._MainThread for t in all_threads)
```

Comment:
I think this might actually not work.

In py3's threading there is a comment that main thread is marked as not alive only very close to interpreter's death, which leads me to conclusion that we might not know when exactly main thread is removed from `threading.enumerate()`.

However, even if that is true, we're not making it worse and we're close to ending support for python 2.7 anyway

Reply:
In all Pythons (at least 2.7 - 3.8), there is a `threading._shutdown` function. That's where the the main thread is stopped - it happens just before waiting for all non-daemon threads.

However, it was a good catch with `enumerate()`. I added checking for alive-ness of the thread.